src/lib/ngsiNotify/Notifier.cpp

```diff
@@ -140,7 +154,15 @@ static bool templateNotify
     std::string key   = it->first;
     std::string value = it->second;
 
+    macroSubstitute(&key,   it->first, ce);
     macroSubstitute(&value, it->second, ce);
+
+    if (key == "")
```

Comment:
Why not add `value` to the `if`as well?
If empty, the key-value will be removed anyway, but later on ...
Like this it's a bit confusing